src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -1906,6 +1906,13 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             ObligationCauseCode::BuiltinDerivedObligation(ref data) => {
                 let parent_trait_ref = self.resolve_vars_if_possible(&data.parent_trait_ref);
                 let ty = parent_trait_ref.skip_binder().self_ty();
+                if let ty::Adt(_, substs) = ty.kind {
+                    if substs.types().any(|t| matches!(t.kind, ty::Error(_))) {
+                        err.cancel();
+                        return;
+                    }
+                }
```

Comment:
```suggestion
                let parent_trait_ref = self.resolve_vars_if_possible(&data.parent_trait_ref);
                if parent_trait_ref.references_error() {
                    err.cancel();
                    return;                
                }
```

`references_error` uses a bitmask that is kept up to date, instead of visiting every type, so it is much faster.